lua/plenary/path.lua

```diff
@@ -515,30 +515,62 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+---Rename this file or directory to the provided path (`opts.new_name`),
+---returning a new Path instance upon success.  The rename is aborted if the
+---new path already exists. Relative paths are interpreted relative to the
+---current working directory.
+---@generic T: Path
+---@param opts { new_name: Path|string }
+---@return T
```

Comment:
There belongs 1. a type and 2. a description after the type to T. Consider installing lua_lsp to get emmylua completions.

Reply:
Oh which lsp is that? I'm currently using [lua-language-server](https://github.com/LuaLS/lua-language-server) (i.e. `require'lspconfig'.lua_ls`). I see I messed up here, and generics aren't well-supported with my lsp anyway, so I'd change it to this:

```suggestion
---@param opts { new_name: Path|string } options table containing the new name
---@return Path # Path representing the new name
```

this is how my lsp says to document table params. I'd also remove the line above declaring the generic.

If this project uses a different lsp, or you just want me to change it to be like the other types in this file, just let me know and I will.

Reply:
> If this project uses a different lsp, or you just want me to change it to be like the other types in this file, just let me know and I will.

No, the one you mentioned is the one I meant. Next time I'll just link to it. The change looks better.